apis/api/commontypes/astarte_common_types.go

```diff
@@ -618,6 +621,12 @@ type AstarteSpec struct {
 	Components AstarteComponentsSpec `json:"components"`
 	// +optional
 	AstarteSystemKeyspace AstarteSystemKeyspaceSpec `json:"astarteSystemKeyspace,omitempty"`
+	// ManualMaintenanceMode pauses all reconciliation activities but still computes the resource
+	// status. It should be used only when the managed Astarte resources requires manual intervention
+	// and the Operator cannot break out of the problem by itself. Do not set this field unless you
+	// know exactly what you are doing.
+	// +optional
+	ManualMaintenanceMode *bool `json:"manualMaintenanceMode,omitempty"`
```

Comment:
Here we could use kubebuilder defaults instead of a pointer. Since the zero value for `bool` is `false`, the `omitempty` tag makes sure the field is not displayed in the CR when `manualManteinanceMode` is set to `false`.
See [this example](https://go2goplay.golang.org/p/ccmG-khsJzD). This allows to remove pointy from the rest of the PR.